testhelper/docker/resource/mysql/mysql.go

```diff
@@ -0,0 +1,83 @@
+package mysql
+
+import (
+	"database/sql"
+	_ "encoding/json"
+	"fmt"
+
+	_ "github.com/go-sql-driver/mysql" // mysql driver
+	"github.com/ory/dockertest/v3"
+	dc "github.com/ory/dockertest/v3/docker"
+
+	"github.com/rudderlabs/rudder-go-kit/testhelper/docker/resource"
+)
+
+const (
+	mysqlDefaultDB       = "sources"
+	mysqlDefaultUser     = "root"
+	mysqlDefaultPassword = "password"
+)
+
+type Resource struct {
+	DB       *sql.DB
+	DBDsn    string
+	Database string
+	Password string
+	User     string
+	Host     string
+	Port     string
+}
+
+func Setup(pool *dockertest.Pool, d resource.Cleaner, opts ...func(*Config)) (*Resource, error) {
+	c := &Config{
+		Tag: "8.2",
+	}
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	// pulls an image, creates a container based on it and runs it
+	mysqlContainer, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "mysql",
+		Tag:        c.Tag,
+		Env: []string{
+			"MYSQL_ROOT_PASSWORD=" + mysqlDefaultPassword,
+			"MYSQL_DATABASE=" + mysqlDefaultDB,
+		},
+	}, func(hc *dc.HostConfig) {
+		hc.ShmSize = c.ShmSize
+	})
+	if err != nil {
+		return nil, err
```

Comment:
Would adding more context would be helpful?
```suggestion
		return nil, fmt.Errorf("could not start resource: %s", err)
```